README.md

```diff
@@ -1376,6 +1392,32 @@ bucket. This will automatically delete files in your bucket that are older than
 permits.
 [You can read more about configuring a retention policy in S3 here.](https://docs.aws.amazon.com/AmazonS3/latest/userguide/how-to-set-lifecycle-configuration-intro.html)
 
+## Document Download
+
+Form-flow library allows users to either:
+
+1. Download individual files
+2. Download a zip folder with all the files associated with a submission.
+
+In order to download from any endpoint, the http session must have an attribute of "id" that matches
+the submission's UUID that you would like to download. There are
+
+### Downloading individual files
+
+To downlodoad individual files a user must reach single-file download
+endpoint: `/file-download/{submissionId}/{fileId}`.  
+The `submissionId` must be included in the session as an attribute and the `fileId` must be found in
+the `user_files` table in order for the file to be retrieved.
+
+### Downloading all files
+
+In order to download all the files associated with a submission a user needs to use the download-all
+endpoint:  `/file-download/{submissionId}`. The download-all endpoint requires that the `id`
+attribute for the httpsession matches the `submissionId` for the group of files in the submission
+that
+you would like to download. When you reach the download-all endpoint every file in the `user_files`
+table associated with a submission are zipped together and downloaded.
```

Comment:
"When you use", versus "reach" might flow better.   "reach" isn't a word I've heard applied to API's much.